### PR TITLE
feat: add `Series.dtype` return annotation

### LINKED ISSUE #1047: [Enh]: Export `nw.DType` from `Series`'s `.dtype` property
### We would like to learn about your use case. For example, if this feature is needed  to adopt Narwhals in an open source project, could you please enter the link to it below?

posit-dev/py-shiny

### Please describe the purpose of the new feature or describe the problem to solve.

Ex: https://github.com/narwhals-dev/narwhals/blob/3f1619da7fc7f8f389e4d00797d029196b9a807b/narwhals/series.py#L305

Current code has to type the series object before I can ask if it is a numeric column. I'd like to be able to drop the `: nw.DType`. 

```python
def column_is_numeric(nw_data: DataFrame[Any], column_index: int) -> bool:
    series_dtype: DType = nw_data[:, column_index].dtype
    return series_dtype.is_numeric()
```

I'm fine with it staying as is... I'm just curious as to the motivation of the `Any` type given the amazing typing everywhere else.

### Suggest a solution if possible.

Return `nw.DType` in the base series class. (Which would require downstream updates)

### If you have tried alternatives, please describe them below.

Type case the `Any` to `nw.DType`

### Additional information that may help us understand your needs.

_No response_

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Closes #1047

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/series.py [hunk 1]
```diff
@@ -302,7 +303,7 @@ def len(self) -> int:
         return len(self._compliant_series)
 
     @property
-    def dtype(self) -> Any:
+    def dtype(self: Self) -> DType:
```
Comment (schloerke) [inline narwhals/series.py:306]: Thank you!

File: narwhals/_arrow/series.py [hunk 2]
```diff
@@ -359,7 +359,7 @@ def alias(self, name: str) -> Self:
         )
 
     @property
-    def dtype(self) -> DType:
+    def dtype(self: Self) -> DType:
```
Comment (schloerke) [inline narwhals/_arrow/series.py:362]: I don't believe it's needed

```suggestion
    def dtype(self) -> DType:
```

But if it is... Can you please teach me why? Thank you

Comment (FBruzzesi) [reply narwhals/_arrow/series.py:362]: I took this as opportunity to go a bit banana into hints for the rest of the codebase (uncommitted here). When we know that `self: Self`, then `self._native_series` is known to be a `pa.ChunkedArray`, which has a `type` property.

Probably some editors do not need it anyway 😇

Comment (MarcoGorelli) [reply narwhals/_arrow/series.py:362]: sounds like something which editors should fix 😄 but in the meantime, if this helps the development workflow, then it seems like a good idea to add it 👍

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi !

**Pull request merged**